### PR TITLE
2021-08-31 commit

### PR DESCRIPTION
**Pull request by rodrigodagostino-globant**
Started the main structure of the blog landing page, creating the `BlogSection` and `BlogPost` components along with basic styles.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -32,3 +32,12 @@ yarn-error.log*
...
 .vercel
+
+# Editor directories and files
+.idea
+.vscode
+*.suo
+*.ntvs*
+*.njsproj
+*.sln
+*.sw?
```
Comment (g-azambrano) [inline .gitignore]: Here there's no empty line at the end of the file. 

You may want to configure Prettier to do this for you in a [pre-commit hook](https://prettier.io/docs/en/precommit.html) or similar. 🤔 

This is not too important, just wanted to point it out!

File: components/BlogPost.module.css [hunk 2]
```diff
@@ -0,0 +1,20 @@
+.blog-post {
+}
+
+.blog-post__media {
+}
+
+.blog-post__content {
+}
```
Comment (g-azambrano) [inline components/BlogPost.module.css]: If you don't need it, you should remove it. 😛

File: components/BlogPost.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import Image from 'next/image';
+
+import styles from './BlogPost.module.css';
+
+type BlogPostProps = {
```
Comment (g-azambrano) [inline components/BlogPost.tsx]: This is fine, but usually, in our project, we tend to use interfaces over types.

The difference is very subtle, so this is not an issue at all!

File: components/BlogSection.module.css [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  margin: 4rem 0;
+}
+
+.blog-section__content {
+  display: flex;
+  gap: 3rem;
+}
+
+.blog-section__content > .blog-section__column {
+}
```
Comment (g-azambrano) [inline components/BlogSection.module.css]: Delete if not needed

File: components/BlogSection.tsx [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+  datePublished: '2021-08-31',
+  featuredImageUrl:
+    'https://images.unsplash.com/photo-1630359486912-4b085e38f1e7?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=748&q=80',
+  author: 'Juan Pérez',
+};
+
+const BlogSection = ({ title, variation }: BlogSectionProps) => {
+  console.log(variation);
+  return (
+    <div className={`${styles['blog-section']} ${styles[`blog-section--${variation}`]}`}>
```
Comment (g-azambrano) [inline components/BlogSection.tsx]: We discussed this on our call earlier, but see how this JSX is very long and nested.

The best we should look after when writing functions in JS (This applies to React components being a function) is to keep them small and doing one thing only.

A pattern we should try to stick to as much as possible.

**Review by g-azambrano — APPROVED** [review]
This PR is looking great especially for just 1-day progress!

Feel free to merge it. In your next PR, you could look into improving the code based on my comments. 😄

File: .gitignore (hunk 1 above)
Comment (rodrigodagostino-globant) [reply .gitignore]: Thanks for the tip! I’ll look into it :)

File: components/BlogPost.module.css (hunk 2 above)
Comment (rodrigodagostino-globant) [reply components/BlogPost.module.css]: Got it! ;)

File: components/BlogPost.tsx (hunk 3 above)
Comment (rodrigodagostino-globant) [reply components/BlogPost.tsx]: I was just thinking about that this morning, and I realized that I don’t really know the difference between those two, so I will look into it and change types for interfaces.

File: components/BlogSection.module.css (hunk 4 above)
Comment (rodrigodagostino-globant) [reply components/BlogSection.module.css]: Got it! ;)

File: components/BlogSection.tsx (hunk 5 above)
Comment (rodrigodagostino-globant) [reply components/BlogSection.tsx]: Yeah, definitely. I will keep that in mind ;)

**Pull request merged**